sdcm/utils/compaction_ops.py

```diff
@@ -0,0 +1,103 @@
+import time
+from typing import Callable, Optional, NamedTuple, Union
+
+from fabric.runners import Result
+
+from sct import LOGGER
+from sdcm.cluster import BaseNode, BaseCluster, BaseScyllaCluster
+from sdcm.rest.storage_service_client import StorageServiceClient
+
+
+class ScrubModes(NamedTuple):
+    ABORT: str = "ABORT"
+    SKIP: str = "SKIP"
+    SEGREGATE: str = "SEGREGATE"
+    VALIDATE: str = "VALIDATE"
+
+
+class NodetoolCommands(NamedTuple):
+    flush: str = "flush"
+    stop_major_compaction: str = "stop COMPACTION"
+    stop_scrub_compaction: str = "stop SCRUB"
+    stop_cleanup_compaction: str = "stop CLEANUP"
+    stop_upgrade_compaction: str = "stop UPGRADE"
+    stop_reshape_compaction: str = "stop RESHAPE"
+
+
+class CompactionOps:
+    NODETOOL_CMD = NodetoolCommands()
+    SCRUB_MODES = ScrubModes()
+
+    def __init__(self, cluster: Union[BaseCluster, BaseScyllaCluster], node: Optional[BaseNode] = None):
+        self.cluster = cluster
+        self.node = node if node else self.cluster.nodes[0]
+        self.storage_service_client = StorageServiceClient(node=self.node)
+
+    def trigger_major_compaction(self, keyspace: str = "keyspace1", cf: str = "standard1") -> Result:
+        return self.storage_service_client.compact_ks_cf(keyspace=keyspace, cf=cf)
+
+    def trigger_scrub_compaction(self,
+                                 keyspace: str = "keyspace1",
+                                 cf: str = "standard1",
+                                 scrub_mode: Optional[str] = None) -> Result:
+        params = {"keyspace": keyspace, "cf": cf}
+
+        if scrub_mode:
```

Comment:
I see the same dict in the `scrub_ks_cf`. Why do we need to create the dict here?

Reply:
Good catch, I'll remove the `if` from here, as it's handled in the client.